plugins/modules/aci_rest.py

```diff
@@ -447,19 +469,15 @@ def main():
             payload = config_object.read()
 
     # Validate payload
-    if rest_type == "json":
-        if content and isinstance(content, dict):
-            # Validate inline YAML/JSON
-            add_annotation(annotation, payload)
-            payload = json.dumps(payload)
-        elif payload and isinstance(payload, str) and HAS_YAML:
+    if rest_type == "json" and (content or payload):
+        if (isinstance(content, str) or isinstance(payload, str)) and HAS_YAML:
```

Comment:
is the or needed? since only payload is used and content is store in payload in 465

Reply:
Changed the if condition.